### PR TITLE
Synced recent updates to #eaedd4c

### PR DESCRIPTION
**Pull request by Jinjiang**
update:
ref: https://github.com/vuejs/vuejs.org/compare/dee0bfd...eaedd4c
<s>ref: https://github.com/vuejs/vuejs.org/compare/dee0bfd...860e624</s>

@yyx990803 hope you have a look at the changes in `sidebar.ejs` and `sponsors.ejs` because there are local sponsors so I couldn't copy the same directly from English website.

Here are some screenshots:

Sidebar:

[en] sidebar with one additional sponsor link

[cn] sidebar without any update because the content is dynamically generated

Sponsors:

[en] sponsors

[cn] sponsors part 1 (Patreon)

[cn] sponsors part 2 (OpenCollective)

### CODE REVIEW

**Comment by yyx990803** [comment]
1. Yes, we keep the sidebar sponsors separate between CN and International versions.

2. On frontpage we stack them with CN sponsor on top.

3. Open Collective recently had some bugs and they just got fixed in the main vuejs.org repo - maybe sync again.

File: src/v2/guide/typescript.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ order: 404
 }
 ```
 
-这句选项告诉 TypeScript 不要处理 ES 模块引入语句 (译注：import .. from ..)。这样 webpack 2 就可以充分利用其基于 ES 模块的 tree-shaking (译注一种在抽象语法树中减除未被使用的死代码的优化技术，简称`摇树优化`)。
+注意你需要引入 `strict: true` (或者至少 `noImplicitThis: true`，这是 `strict` 模式的一部分) 以影响组件方法中 `this` 的类型检查，否则它会始终被看作 `any` 类型。
```
Comment (Justineo) [inline src/v2/guide/typescript.md]: 引入……以*利用*组件方法中 `this` 的类型检查

File: src/v2/guide/typescript.md [hunk 2]
```diff
@@ -152,3 +143,37 @@ var vm = new Vue({
   myOption: 'Hello'
 })
 ```
+
+## 注明返回值
+
+因为 Vue 的声明文件的循环的本源，TypeScript 可能在推断某个方法的类型的时候存在困难。因此，你可能需要在 `render` 或 `computed` 里的方法上注明返回值。
```
Comment (Justineo) [inline src/v2/guide/typescript.md]: “因为 Vue 的声明文件天生就具有循环性”？

File: src/v2/guide/typescript.md [hunk 3]
```diff
@@ -152,3 +143,37 @@ var vm = new Vue({
   myOption: 'Hello'
 })
 ```
+
+## 注明返回值
```
Comment (Justineo) [inline src/v2/guide/typescript.md]: annotate/annotation 统一为“标注”吧？

File: src/v2/guide/typescript.md [hunk 4]
```diff
@@ -152,3 +143,37 @@ var vm = new Vue({
...
+import Vue, { VNode } from 'vue'
+
+const Component = Vue.extend({
+  data () {
+    return {
+      msg: 'Hello'
+    }
+  },
+  methods: {
+    // 需要注明有 `this` 参与运算的返回值类型
```
Comment (Justineo) [inline src/v2/guide/typescript.md]: 标注？

File: src/v2/guide/typescript.md [hunk 5]
```diff
@@ -152,3 +143,37 @@ var vm = new Vue({
...
+    greet (): string {
+      return this.msg + ' world'
+    }
+  },
+  computed: {
+    // 需要注明
```
Comment (Justineo) [inline src/v2/guide/typescript.md]: 同上

File: src/v2/guide/typescript.md (hunk 2 above)
Comment (Jinjiang) [reply src/v2/guide/typescript.md]: 靠谱

**Comment by yyx990803** [comment]
LGTM

**Pull request merged**